contract_manager/scripts/deploy_evm_pricefeed_contracts.ts

```diff
@@ -0,0 +1,298 @@
+    },
+    "save-contract": {
+      type: "boolean",
+      demandOption: false,
+      default: true,
+      desc: "Save the contract to the store",
+    },
+  });
+
+async function deployIfNotCached(
+  chain: EvmChain,
+  config: DeploymentConfig,
+  artifactName: string,
+  deployArgs: any[] // eslint-disable-line  @typescript-eslint/no-explicit-any
+): Promise<string> {
+  const cache = existsSync(CACHE_FILE)
+    ? JSON.parse(readFileSync(CACHE_FILE, "utf8"))
+    : {};
+
+  const cacheKey = `${chain.getId()}-${artifactName}`;
+  if (cache[cacheKey]) {
+    const address = cache[cacheKey];
+    console.log(
+      `Using cached deployment of ${artifactName} on ${chain.getId()} at ${address}`
+    );
+    return address;
+  }
+
+  const artifact = JSON.parse(
+    readFileSync(join(config.jsonOutputDir, `${artifactName}.json`), "utf8")
+  );
+
+  console.log(`Deploying ${artifactName} on ${chain.getId()}...`);
+
+  const addr = await chain.deploy(
+    config.privateKey,
+    artifact["abi"],
+    artifact["bytecode"],
+    deployArgs,
+    config.gasMultiplier,
+    config.gasPriceMultiplier
+  );
+
+  console.log(`✅ Deployed ${artifactName} on ${chain.getId()} at ${addr}`);
+
+  cache[cacheKey] = addr;
+  writeFileSync(CACHE_FILE, JSON.stringify(cache, null, 2));
+  return addr;
+}
+
+function getWeb3Contract(
+  config: DeploymentConfig,
+  artifactName: string,
+  address: string
+): Contract {
+  const artifact = JSON.parse(
+    readFileSync(join(config.jsonOutputDir, `${artifactName}.json`), "utf8")
+  );
+  const web3 = new Web3();
+  return new web3.eth.Contract(artifact["abi"], address);
+}
+
+async function deployWormholeReceiverContracts(
+  chain: EvmChain,
+  config: DeploymentConfig
+): Promise<string> {
+  const receiverSetupAddr = await deployIfNotCached(
+    chain,
+    config,
+    "ReceiverSetup",
+    []
+  );
+
+  const receiverImplAddr = await deployIfNotCached(
+    chain,
+    config,
+    "ReceiverImplementation",
+    []
+  );
+
+  // Craft the init data for the proxy contract
+  const setupContract = getWeb3Contract(
+    config,
+    "ReceiverSetup",
+    receiverSetupAddr
+  );
+
+  const { wormholeConfig } = getDefaultDeploymentConfig(config.type);
+
+  const initData = setupContract.methods
+    .setup(
+      receiverImplAddr,
+      wormholeConfig.initialGuardianSet.map((addr: string) => "0x" + addr),
+      chain.getWormholeChainId(),
+      wormholeConfig.governanceChainId,
+      "0x" + wormholeConfig.governanceContract
+    )
+    .encodeABI();
+
+  const wormholeReceiverAddr = await deployIfNotCached(
+    chain,
+    config,
+    "WormholeReceiver",
+    [receiverSetupAddr, initData]
+  );
+
+  const wormholeEvmContract = new WormholeEvmContract(
+    chain,
+    wormholeReceiverAddr
+  );
+
+  if (config.type === "stable") {
+    console.log(`Syncing mainnet guardian sets for ${chain.getId()}...`);
+    // TODO: Add a way to pass gas configs to this
```

Comment:
Since this is a generic method I can't easily add the gas options to it. Something to refactor later in the future.